sources/common/src/XmlSignalFactory.cpp

```diff
@@ -0,0 +1,175 @@
+/// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+/// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+/// IN THE SOFTWARE.
+
+#include "XmlSignalFactory.h"
+#include "XmlEventSignalCreator.h"
+#include "XmlFSignalCreator.h"
+#include "XmlISignalCreator.h"
+#include "VcdException.h"
+
+PARSER::XmlDescription::XmlDescription(const pugi::xml_node &rNode) :
+        m_RegexLine(),
+        m_Timestamp(),
+        m_Name(),
+        m_Value(),
+        m_Size()
+{
+    for (pugi::xml_node tag : rNode.children())
+    {
+        if (!std::strcmp("line", tag.name()))
+        {
+            m_RegexLine = tag.child_value();
+        }
+
+        if (!std::strcmp("timestamp", tag.name()))
+        {
+            m_Timestamp = tag.child_value();
+        }
+
+        if (!std::strcmp("name", tag.name()))
+        {
+            m_Name = tag.child_value();
+        }
+
+        if (!std::strcmp("value", tag.name()))
+        {
+            m_Value = tag.child_value();
+        }
+
+        if (!std::strcmp("size", tag.name()))
+        {
+            m_Size = tag.child_value();
+        }
+    }
+}
+
+const std::string & PARSER::XmlDescription::GetRegex() const
+{
+    if (m_RegexLine.length())
+    {
+        return m_RegexLine;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_REGEX_EXPRESSION,
+        "XML - No parsing regular expression.");
+}
+
+const std::string & PARSER::XmlDescription::GetTimestamp() const
+{
+    if (m_Timestamp.length())
+    {
+        return m_Timestamp;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_TIMESTAMP_EXPRESSION,
+        "XML - No timestamp expression.");
+}
+
+const std::string & PARSER::XmlDescription::GetName() const
+{
+    if (m_Name.length())
+    {
+        return m_Name;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_NAME_EXPRESSION,
+        "XML - No name expression.");
+}
+
+const std::string & PARSER::XmlDescription::GetValue() const
+{
+    if (m_Value.length())
+    {
+        return m_Value;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_VALUE_EXPRESSION,
+        "XML - No value expression.");
+}
+
+const std::string & PARSER::XmlDescription::GetSize() const
+{
+    if (m_Size.length())
+    {
+        return m_Size;
+    }
+    throw EXCEPTION::VcdException(EXCEPTION::Error::NO_SIZE_EXPRESSION,
+        "XML - No size expression.");
+}
+
+PARSER::XmlSignalFactory::XmlSignalFactory(const std::string& rXmlFileName) : SignalFactory(false)
+{
+    // Parse the XML and add signal creators.
+    // pugi test
+    pugi::xml_document doc;
+    if (!doc.load_file(rXmlFileName.c_str())) throw (1);  /// @todo Fix filename and error
```

Comment:
Do we want to have such todos on main branch?

Reply:
Definitely not ;)